pkg/config/nodetreemodel/getter.go

```diff
@@ -0,0 +1,234 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package nodetreemodel
+
+import (
+	"maps"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/mohae/deepcopy"
+	"github.com/spf13/cast"
+	"golang.org/x/exp/slices"
+)
+
+// GetKnownKeysLowercased returns all the keys that meet at least one of these criteria:
+// 1) have a default, 2) have an environment variable binded or 3) have been SetKnown()
+// Note that it returns the keys lowercased.
+func (c *ntmConfig) GetKnownKeysLowercased() map[string]interface{} {
+	c.RLock()
+	defer c.RUnlock()
+
+	// GetKnownKeysLowercased returns a fresh map, so the caller may do with it
+	// as they please without holding the lock.
+	ret := make(map[string]interface{})
+	for key, value := range c.knownKeys {
+		ret[key] = value
+	}
+	return ret
+}
+
+// GetEnvVars gets all environment variables
+func (c *ntmConfig) GetEnvVars() []string {
+	c.RLock()
+	defer c.RUnlock()
+	vars := make([]string, 0, len(c.configEnvVars))
+	for v := range c.configEnvVars {
+		vars = append(vars, v)
+	}
+	return vars
+}
+
+// GetProxies returns the proxy settings from the configuration
+func (c *ntmConfig) GetProxies() *model.Proxy {
+	c.Lock()
+	hasProxies := c.proxies
+	c.Unlock()
+	if hasProxies != nil {
+		return hasProxies
+	}
+	if c.GetBool("fips.enabled") {
+		return nil
+	}
+	if !c.IsSet("proxy.http") && !c.IsSet("proxy.https") && !c.IsSet("proxy.no_proxy") {
+		return nil
+	}
+	p := &model.Proxy{
+		HTTP:    c.GetString("proxy.http"),
+		HTTPS:   c.GetString("proxy.https"),
+		NoProxy: c.GetStringSlice("proxy.no_proxy"),
+	}
+	c.proxies = p
```

Comment:
Just noticed, this field isn't protected by a Lock. Unfortunately, we can't keep the lock over the entire function because `GetBool` and `IsSet` etc also use the lock.